neo/Network/P2P/RemoteNode.cs

```diff
@@ -135,7 +135,10 @@ protected override void OnReceive(object message)
         private void OnPingPayload(PingPayload payload)
         {
             if (payload.LastBlockIndex > LastBlockIndex)
+            {
                 LastBlockIndex = payload.LastBlockIndex;
+                system.TaskManager.Tell(new TaskManager.Update { LastBlockIndex = LastBlockIndex });
```

Comment:
Is this correct?
Because the sender of `ping` sends his `Blockchain.Singleton.Height`.
The same should be the response, no?

Reply:
I use the style from https://github.com/neo-project/neo/blob/35e1d17ae1b989d03a835498f2fc6f968c120f43/neo/Network/P2P/RemoteNode.cs#L174

Reply:
I mean the logic itself, because the way it is the `receiver` is responding with the same value the payload carried, does it make sense?

Reply:
I think it should be `system.TaskManager.Tell(new TaskManager.Update { LastBlockIndex = Blockchain.Singleton.Height })`

Reply:
> I think it should be `system.TaskManager.Tell(new TaskManager.Update { LastBlockIndex = Blockchain.Singleton.Height })`

This version use own block height instead of the remote block height.

Reply:
Yes, we should assign the remote block height: `LastBlockIndex = payload.LastBlockIndex;`,
but reply with the own block height, yes?

Reply:
It's already a response.

Reply:
I got it. Sorry for the delay.

The `ProtocolHandler is replaying with the `Pong`:

https://github.com/neo-project/neo/blob/35e1d17ae1b989d03a835498f2fc6f968c120f43/neo/Network/P2P/ProtocolHandler.cs#L312-L316